stan/math/prim/prob/discrete_range_cdf.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef STAN_MATH_PRIM_PROB_DISCRETE_RANGE_CDF_HPP
+#define STAN_MATH_PRIM_PROB_DISCRETE_RANGE_CDF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/inv.hpp>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/max_size.hpp>
+#include <stan/math/prim/fun/size_zero.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/** \ingroup prob_dists
+ * Return the CDF of a discrete range distribution for the given y,
+ * lower and upper bounds (all integers).
+ *
+ * `y`, `lower` and `upper` can each be a scalar or a one-dimensional container.
+ * Any container arguments must be the same size.
+ *
+ * @tparam T_y type of scalar, either int or std::vector<int>
+ * @tparam T_lower type of lower bound, either int or std::vector<int>
+ * @tparam T_upper type of upper bound, either int or std::vector<int>
+ *
+ * @param y integer random variable
+ * @param lower integer lower bound
+ * @param upper integer upper bound
+ * @return The CDF evaluated at the specified arguments. If containers are
+ * supplied, returns the product of the CDFs.
+ * @throw std::domain_error if upper is smaller than lower.
+ * @throw std::invalid_argument if non-scalar arguments are of different
+ * sizes.
+ */
+template <typename T_y, typename T_lower, typename T_upper>
+double discrete_range_cdf(const T_y& y, const T_lower& lower,
+                          const T_upper& upper) {
+  static const char* function = "discrete_range_cdf";
+  if (size_zero(y, lower, upper)) {
```

Comment:
This should come after the consistent size checks.  It should be illegal to call with mismatched arguments even one of the sizes is zero.